Flabbergasted-Flamingos/bangazon-api
# Kf training program access

# Description
Added model, serializer, view, and url for the trainings resource

## Number of Fixes


## Related Ticket(s)
#10 

## Problem to Solve
Allows developers to get, put, post, and delete from the training resource. Delete only if the training is in the future.

## Proposed Changes
Allows developers to get, put, post, and delete from the training resource. Delete only if the training is in the future.

## Expected Behavior
Should be able to post, put, get, and delete from Trainings

## Steps to Test Solution

```
git fetch --all
git checkout KFTrainingProgramAccess
```
```
rm -rf api/migrations
rm db.sqlite3
```
```
python manage.py makemigrations api
```
```
python manage.py migrate api
```
1. Attempt to get, post, put
1. Make sure past events cannot be deleted
1. delete future event


## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
